src/frontend/src/modules/Client/Devices.vue

```diff
@@ -68,6 +74,24 @@ export default {
     setMapCenter(device) {
       EventBus.$emit("setMapCenterForDevice", device)
     },
+    navigateToDeviceDetail(device) {
+      let route = ""
+
+      switch (device.device_type) {
+        case "meter":
+          route = `/meters/${device.device_serial}`
+          break
+        case "solar_home_system":
+          route = `/solar-home-systems/${device.id}`
+          break
+        case "e_bike":
+          route = `/e-bikes/${device.id}`
+          break
+        default:
+          route = `/devices/${device.id}`
```

Comment:
Rather than "hardcoding" the routes here, could you pull these routes in via the corresponding `repositories`? For example https://github.com/EnAccess/micropowermanager/blob/main/src/frontend/src/repositories/SolarHomeSystemRepository.js#L10-L12 for SHS. This would be more consistent with how the rest of the frontend is structured.